### DISCUSSION TITLE
Option to specificy nextjs cacheOnNavigation cache name?

### DISCUSSION THREAD

**Discussion by ZackTRobertson** [serwist/serwist#178] [Q&A] [answered]
As the title suggests, we can currently catch Next navigations and cache them using the cacheOnNavigation option in the next.config.mjs file. Currently, this caches these requests by default in a cache titled 'pages'. Is there an option to change the default name so that it can be adjusted for each new install? I've dug around the docs but can't seem to find anything.

Thanks again for any advice on this matter!

EDIT:
I do realize we can catch the request.mode as 'navigate' in a matcher function, but I was hoping there would be a simple config option available :)

**Comment by DuCanhGH** [comment, marked as answer]
@ZackTRobertson 

> Is there an option to change the default name so that it can be adjusted for each new install?

There's no way as of right now, unfortunately :( You will need to disable `cacheOnNavigation` and... reimplement the feature yourself. This can be done by overridding `history.pushState` and `history.replaceState` and adding a `online` event listener:

```ts
useEffect(() => {
  const cacheOnNavigation = async (url?: string | URL | null | undefined) => {
    if (!window.navigator.onLine || !url) {
      return;
    }
    window.serwist.messageSW({
      type: 'CACHE_URLS',
      payload: { urlsToCache: [url] },
    });
  };

  const pushState = history.pushState;
  history.pushState = (...args) => {
    pushState.apply(history, args);
    cacheOnNavigation(args[2]);
  };

  const replaceState = history.replaceState;
  history.replaceState = (...args) => {
    replaceState.apply(history, args);
    cacheOnNavigation(args[2]);
  };

  window.addEventListener("online", () => {
    cacheOnNavigation(window.location.pathname);
  });
}, []);
```

This differs from `@serwist/next`'s implementation in that it uses `Serwist.messageSW` rather than sending a message to a custom worker. Arguably, we can do the same, but that may incur a breaking change. Should've done this earlier, I'd argue. We also can't ensure that the service worker handles this message, unfortunately. Perhaps I should rethink this feature in v10 :) I really think `cacheOnNavigation` is a mistake in `next-pwa`, `@ducanh2912/next-pwa`, and `@serwist/next`, but I haven't got around to reimagining this yet.

**Reply by ZackTRobertson** [reply]
Thanks for the detailed reply. It's not a pressing issue, but as you say it could be something to address in a future release.
